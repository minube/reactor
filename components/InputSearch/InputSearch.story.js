import React, { PureComponent } from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import {
  text, boolean, color, object,
} from '@storybook/addon-knobs/react';
import InputSearch from './InputSearch';

const DATA = [
  {name: 'Madrid', id: 23},
  {name: 'Madrid Snowzone', id: 12},
  {name: 'Comunidad Madrid', id: 65},
  {name: 'San Lorenzo del escorial', id: 76},
  {name: 'Alcalá de Henares', id: 132},
  {name: 'Aranjuez', id: 73},
];

const STYLE = { padding: 10, width: 256 };

_onChange = () => DATA;
_onClickItem = () => {};

class HOC extends PureComponent {
  constructor(props) {
    super(props);
    this.state = { data: props.data };
  }

  _onChange = () => {
    const { onChange } = this.props;
    this.setState({ data: onChange() });
  }

  render() {
    const { _onChange, props, state: { data } } = this;
    return <InputSearch {...props} dataSource={data} onChange={_onChange} />;
  }
}

storiesOf('✅ InputSearch', module)
  .add('default', () => (
    <InputSearch />
  ))
  .add('color', () => (
    <InputSearch color="green" />
  ))
  .add('dataSource', () => (
    <InputSearch dataSource={DATA} />
  ))
  .add('disabled', () => (
    <InputSearch disabled />
  ))
  .add('hint', () => (
    <InputSearch hint="Optional field" />
  ))
  .add('label', () => (
    <InputSearch label="InputSearch" />
  ))
  .add('placeholder', () => (
    <InputSearch placeholder={"Looking for..."} />
  ))
  .add('value', () => (
    <InputSearch value="Madrid" />
  ))
  .add('onChange', () => (
    <HOC onChange={_onChange} dataSource={this.data} />
  ))
  .add('onClickItem', () => (
    <HOC onChange={_onChange} onClickItem={_onClickItem} dataSource={this.data} />
  ))
  .add('🏀 Playground', () => (
    <InputSearch
      color={color('color', undefined)}
      hint={text('hint', 'This is a hint')}
      disabled={boolean('disabled', false)}
      label={text('label', 'This is a label')}
      value={text('value', 'Madrid')}
      onChange={action('InputSearch.onChange()')}
      style={object('style', STYLE)}
    />
  ));
