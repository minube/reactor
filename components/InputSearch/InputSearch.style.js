import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const {
  BORDER_RADIUS, COLOR, UNIT, SPACE,
} = THEME;

const HEIGHT = UNIT * 4.4;
const HEIGHT_LABEL = UNIT * 6.8;

export default StyleSheet.create({
  input: {
    width: '100%',
  },

  item: {
    paddingVertical: SPACE.MEDIUM,
  },

  border: {
    borderBottomWidth: 1,
    borderColor: COLOR.BASE,
  },

  list: {
    backgroundColor: COLOR.WHITE,
    borderRadius: BORDER_RADIUS,
    paddingHorizontal: SPACE.MEDIUM,
    position: 'absolute',
    zIndex: 2,
    width: '100%',
  },

  marginList: {
    top: HEIGHT,
  },

  marginListLabel: {
    top: HEIGHT_LABEL,
  },
});
