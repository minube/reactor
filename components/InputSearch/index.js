export { default } from './InputSearch'; // eslint-disable-line
