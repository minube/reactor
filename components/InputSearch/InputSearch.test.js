import React from 'react';
import renderer from 'react-test-renderer';
import InputSearch from './InputSearch';

const VALUE = 'Madrid';
const DATA = [
  {name: 'Madrid', id: 23},
  {name: 'Madrid Snowzone', id: 12},
];
const FUNC = () => {};
const STYLE = { backgroundColor: 'rgba(0,255,0,0.25)', padding: 10 };

describe('<InputSearch>', () => {
  it('renders', () => {
    const tree = renderer.create(<InputSearch />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {disabled}', () => {
    const tree = renderer.create(<InputSearch disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {dataSource}', () => {
    const tree = renderer.create(<InputSearch dataSource={DATA} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {hint}', () => {
    const tree = renderer.create(<InputSearch hint="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {label}', () => {
    const tree = renderer.create(<InputSearch label="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {value}', () => {
    const tree = renderer.create(<InputSearch value={VALUE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onChange}', () => {
    const tree = renderer.create(<InputSearch onChange={FUNC} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onClickItem}', () => {
    const tree = renderer.create(<InputSearch onClickItem={FUNC} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {placeholder}', () => {
    const tree = renderer.create(<InputSearch placeholder={"Hello World"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<InputSearch style={STYLE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
