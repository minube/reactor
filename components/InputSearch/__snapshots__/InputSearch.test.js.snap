// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSearch> renders 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 3,
          "paddingHorizontal": 16,
          "position": "absolute",
          "width": "100%",
          "zIndex": 2,
        },
        Object {
          "top": 44,
        },
      ]
    }
  />
</View>
`;

exports[`<InputSearch> when {dataSource} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 3,
          "paddingHorizontal": 16,
          "position": "absolute",
          "width": "100%",
          "zIndex": 2,
        },
        Object {
          "top": 44,
        },
      ]
    }
  >
    <View
      accessible={true}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="box-only"
      style={
        Array [
          undefined,
          Array [
            Object {
              "paddingVertical": 16,
            },
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#cdd2d4",
            },
          ],
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontFamily": undefined,
              "fontWeight": "400",
              "textAlignVertical": "center",
            },
            Object {
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 18,
            },
            false,
            Object {},
          ]
        }
      >
        Madrid
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 3,
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#cdd2d4",
              "height": 0,
              "opacity": 0,
              "width": 0,
              "zIndex": 0,
            }
          }
        />
      </View>
    </View>
    <View
      accessible={true}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="box-only"
      style={
        Array [
          undefined,
          Array [
            Object {
              "paddingVertical": 16,
            },
            false,
          ],
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontFamily": undefined,
              "fontWeight": "400",
              "textAlignVertical": "center",
            },
            Object {
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 18,
            },
            false,
            Object {},
          ]
        }
      >
        Madrid Snowzone
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 3,
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#cdd2d4",
              "height": 0,
              "opacity": 0,
              "width": 0,
              "zIndex": 0,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`<InputSearch> when {disabled} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          Object {
            "backgroundColor": "#f4f4f4",
          },
          false,
          false,
          false,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={false}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            Object {
              "color": "#707070",
            },
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 3,
          "paddingHorizontal": 16,
          "position": "absolute",
          "width": "100%",
          "zIndex": 2,
        },
        Object {
          "top": 44,
        },
      ]
    }
  />
</View>
`;

exports[`<InputSearch> when {hint} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontWeight": "400",
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 12,
            "lineHeight": 16,
          },
          Object {
            "color": "#707070",
          },
          Object {
            "marginTop": 4,
            "maxWidth": "100%",
            "paddingLeft": 0,
          },
        ]
      }
    >
      Hello World
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 3,
          "paddingHorizontal": 16,
          "position": "absolute",
          "width": "100%",
          "zIndex": 2,
        },
        Object {
          "top": 44,
        },
      ]
    }
  />
</View>
`;

exports[`<InputSearch> when {label} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontWeight": "400",
            "textAlignVertical": "center",
          },
          Object {
            "fontFamily": undefined,
            "fontSize": 14,
            "fontWeight": "600",
            "lineHeight": 16,
          },
          Object {
            "color": "#707070",
          },
          Object {
            "color": "#4a4a4a",
            "marginBottom": 8,
          },
        ]
      }
    >
      Hello World
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 3,
          "paddingHorizontal": 16,
          "position": "absolute",
          "width": "100%",
          "zIndex": 2,
        },
        Object {
          "top": 68,
        },
      ]
    }
  />
</View>
`;

exports[`<InputSearch> when {onChange} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 3,
          "paddingHorizontal": 16,
          "position": "absolute",
          "width": "100%",
          "zIndex": 2,
        },
        Object {
          "top": 44,
        },
      ]
    }
  />
</View>
`;

exports[`<InputSearch> when {onClickItem} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onClickItem={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 3,
          "paddingHorizontal": 16,
          "position": "absolute",
          "width": "100%",
          "zIndex": 2,
        },
        Object {
          "top": 44,
        },
      ]
    }
  />
</View>
`;

exports[`<InputSearch> when {placeholder} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Hello World"
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 3,
          "paddingHorizontal": 16,
          "position": "absolute",
          "width": "100%",
          "zIndex": 2,
        },
        Object {
          "top": 44,
        },
      ]
    }
  />
</View>
`;

exports[`<InputSearch> when {style} 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "rgba(0,255,0,0.25)",
        "padding": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 3,
          "paddingHorizontal": 16,
          "position": "absolute",
          "width": "100%",
          "zIndex": 2,
        },
        Object {
          "top": 44,
        },
      ]
    }
  />
</View>
`;

exports[`<InputSearch> when {value} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../components/Icon/assets/search.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 20,
                "width": 20,
              },
              Object {
                "alignSelf": "flex-start",
                "height": 18,
                "marginRight": 4,
                "width": 18,
              },
              undefined,
            ]
          }
        />
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value="Madrid"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 3,
          "paddingHorizontal": 16,
          "position": "absolute",
          "width": "100%",
          "zIndex": 2,
        },
        Object {
          "top": 44,
        },
      ]
    }
  />
</View>
`;
